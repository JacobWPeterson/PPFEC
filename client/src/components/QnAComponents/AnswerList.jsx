import React from 'react';
import Answer from './Answer.jsx';

const AnswerList = (props) => (
  <div>

    {props.answers.map((answer) => (<Answer answer={answer} key={answer.id} id={answer.id} />))}

  </div>
);

export default AnswerList;
