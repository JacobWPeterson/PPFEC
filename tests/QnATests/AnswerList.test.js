import React from 'react';
import { mount } from 'enzyme';
import AnswerList from '../../client/src/components/QnAComponents/AnswerList.jsx';

describe('This is for the Answers List Component', () => {
  const props = {
    answers: [
      {
        answerer_name: 'Charity18',
        body: 'Eos ipsam totam quia qui voluptate ipsum itaque quaerat.',
        date: '2020-06-05T00:00:00.000Z',
        helpfulness: 15,
        id: 719454,
        photos: [],
      },
      {
        answerer_name: 'Willa_Botsford',
        body: 'Nesciunt aliquam sit consequatur enim reprehenderit.',
        date: '2020-09-20T00:00:00.000Z',
        helpfulness: 10,
        id: 719455,
        photos: [],
      },
    ],
  };

  const answerListComp = mount(<AnswerList {...props} />);

  it('Should render AnswerList Properly', () => {
    expect(answerListComp).toMatchSnapshot();
  });
});
