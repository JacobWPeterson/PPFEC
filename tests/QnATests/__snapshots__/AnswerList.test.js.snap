// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This is for the Answers List Component Should render AnswerList Properly 1`] = `
<AnswerList
  answers={
    Array [
      Object {
        "answerer_name": "Charity18",
        "body": "Eos ipsam totam quia qui voluptate ipsum itaque quaerat.",
        "date": "2020-06-05T00:00:00.000Z",
        "helpfulness": 15,
        "id": 719454,
        "photos": Array [],
      },
      Object {
        "answerer_name": "Willa_Botsford",
        "body": "Nesciunt aliquam sit consequatur enim reprehenderit.",
        "date": "2020-09-20T00:00:00.000Z",
        "helpfulness": 10,
        "id": 719455,
        "photos": Array [],
      },
    ]
  }
>
  <div>
    <Answer
      answer={
        Object {
          "answerer_name": "Charity18",
          "body": "Eos ipsam totam quia qui voluptate ipsum itaque quaerat.",
          "date": "2020-06-05T00:00:00.000Z",
          "helpfulness": 15,
          "id": 719454,
          "photos": Array [],
        }
      }
      id={719454}
      key="719454"
    >
      <div>
        <Answer__Body>
          <div
            className="Answer__Body-sc-3mvxjy-6 gnmbP"
          >
            Eos ipsam totam quia qui voluptate ipsum itaque quaerat.
          </div>
        </Answer__Body>
        <div>
          <Photos
            photos={Array []}
          >
            <div />
          </Photos>
        </div>
        <Answer__BottomContainer>
          <div
            className="Answer__BottomContainer-sc-3mvxjy-0 kfNGYX"
          >
            <Answer__NameDate>
              <div
                className="Answer__NameDate-sc-3mvxjy-1 jcOBlI"
              >
                By Charity18 on  June, 05 2020
              </div>
            </Answer__NameDate>
            |
            <Answer__Helpful>
              <div
                className="Answer__Helpful-sc-3mvxjy-2 iSIJxX"
              >
                Helpful?
              </div>
            </Answer__Helpful>
            <Answer__Yes
              onClick={[Function]}
            >
              <div
                className="Answer__Yes-sc-3mvxjy-3 bpatiV"
                onClick={[Function]}
              >
                Yes
              </div>
            </Answer__Yes>
            <Answer__Help>
              <div
                className="Answer__Help-sc-3mvxjy-4 gOKkij"
              >
                (15)
              </div>
            </Answer__Help>
            |
            <Answer__Report
              onClick={[Function]}
            >
              <div
                className="Answer__Report-sc-3mvxjy-5 fkJrli"
                onClick={[Function]}
              >
                Report
              </div>
            </Answer__Report>
          </div>
        </Answer__BottomContainer>
      </div>
    </Answer>
    <Answer
      answer={
        Object {
          "answerer_name": "Willa_Botsford",
          "body": "Nesciunt aliquam sit consequatur enim reprehenderit.",
          "date": "2020-09-20T00:00:00.000Z",
          "helpfulness": 10,
          "id": 719455,
          "photos": Array [],
        }
      }
      id={719455}
      key="719455"
    >
      <div>
        <Answer__Body>
          <div
            className="Answer__Body-sc-3mvxjy-6 gnmbP"
          >
            Nesciunt aliquam sit consequatur enim reprehenderit.
          </div>
        </Answer__Body>
        <div>
          <Photos
            photos={Array []}
          >
            <div />
          </Photos>
        </div>
        <Answer__BottomContainer>
          <div
            className="Answer__BottomContainer-sc-3mvxjy-0 kfNGYX"
          >
            <Answer__NameDate>
              <div
                className="Answer__NameDate-sc-3mvxjy-1 jcOBlI"
              >
                By Willa_Botsford on  September, 20 2020
              </div>
            </Answer__NameDate>
            |
            <Answer__Helpful>
              <div
                className="Answer__Helpful-sc-3mvxjy-2 iSIJxX"
              >
                Helpful?
              </div>
            </Answer__Helpful>
            <Answer__Yes
              onClick={[Function]}
            >
              <div
                className="Answer__Yes-sc-3mvxjy-3 bpatiV"
                onClick={[Function]}
              >
                Yes
              </div>
            </Answer__Yes>
            <Answer__Help>
              <div
                className="Answer__Help-sc-3mvxjy-4 gOKkij"
              >
                (10)
              </div>
            </Answer__Help>
            |
            <Answer__Report
              onClick={[Function]}
            >
              <div
                className="Answer__Report-sc-3mvxjy-5 fkJrli"
                onClick={[Function]}
              >
                Report
              </div>
            </Answer__Report>
          </div>
        </Answer__BottomContainer>
      </div>
    </Answer>
  </div>
</AnswerList>
`;
